### ISSUE TITLE
False positive ``expression-not-assigned`` in ternary operator launching function without return

### ISSUE THREAD

**Issue by jamesbraza** [PyCQA/pylint#4564]
### Steps to reproduce

Given a file `a.py`:
```python
def none_return_1() -> None:
    pass

def none_return_2() -> None:
    pass

def some_code(flag: bool) -> None:
    none_return_1() if flag else none_return_2()
```

### Current behavior

Result of `pylint a.py`:
```
a.py:8:4:W0106: Expression "none_return_1() if flag else none_return_2()" is assigned to nothing (expression-not-assigned)
```

### Expected behavior

I think this `expression-not-assigned` is a false positive here, since there are no `return` statements in either function.

#### Bad Workaround

To work around, I am just putting a `_ =` out front like so:

```python
def some_code(flag: bool) -> None:
    _ = none_return_1() if flag else none_return_2()
```

However, if one uses the `_ = ` workaround, then mypy will now raise errors:

```
a.py: note: In function "some_code":
a.py:8:9: error: "none_return_1" does not return a value  [func-returns-value]
a.py:8:38: error: "none_return_2" does not return a value  [func-returns-value]
```

So you can't win!

#### Better Workaround

Using a `pylint: disable=expression-not-assigned` in this case.

I feel like this is now "adding extra code to satisfy my code QA tools" which I prefer not to do.

### pylint --version output

Result of `pylint --version` output:

```
pylint 2.8.3
astroid 2.5.6
Python 3.8.8 (default, Apr 22 2021, 12:26:49)
[Clang 12.0.0 (clang-1200.0.32.27)]
```

**Comment by Pierre-Sassoulas** [comment]
I think you could also do this instead:
```python
if flag:
    none_return_1() 
else:
    none_return_2()
```
I agree it's a false positive, but a ternary operation is often used for assignment.

**Comment by mbyrnepr2** [comment]
Closing this one as there is a more recent issue which duplicates this one and also includes a discussion: https://github.com/PyCQA/pylint/issues/8129
